pkg/segment/structs/segstructs.go

```diff
@@ -438,6 +451,26 @@ func (qa *QueryAggregators) HasQueryAggergatorBlockInChain() bool {
 	return false
 }
 
+// To determine whether it contains ValueColRequest
+func (qa *QueryAggregators) HasValueColRequest() bool {
+	for _, agg := range qa.MeasureOperations {
+		if agg.ValueColRequest != nil {
+			return true
+		}
+	}
+	return false
+}
+
+// To determine whether it contains Aggregate Func: Values()
+func (qa *QueryAggregators) HasValuesFunc() bool {
+	for _, agg := range qa.MeasureOperations {
+		if agg.MeasureFunc == utils.Values {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Why do we need to know if it has specifically the `Values()` aggregation?

Reply:
In the `func applyAggOpOnSegments()`, Values func is similar to Eval statements. They all need to recompute the raw data to get the results, since segment only store statistical data such as min, max... 

So i use this condition to judge if current QueryAggregators should be recomputed or not

Reply:
Ok that makes sense. Can we not reuse whatever logic we have for computing the `Cardinality` but get the actual values rather than just the number of values?